exploration_scripts/trajectory_plotting

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python
+from matplotlib import pyplot as plt
+import numpy as np
+import rosbag
+import sys
+
+def get_waypoints(bag_name):
+    bag = rosbag.Bag(bag_name, 'r')
+    waypoints = None
+    for topic, msg, t in bag.read_messages():
+        print "Number of waypoints: ", len(msg.goal.trajectory.joint_trajectory.points)
+        waypoints = msg.goal.trajectory.joint_trajectory.points
+    bag.close()
+
+    positions = np.array([np.array(wp.positions) for wp in waypoints])
+    velocities = np.array([np.array(wp.velocities) for wp in waypoints])
+    accelerations = np.array([np.array(wp.accelerations) for wp in waypoints])
+    times = np.array([wp.time_from_start.to_sec() for wp in waypoints])
+    return (positions, velocities, accelerations, times)
+
+def calc_derivative_from_sequence(sequence, times):
+    derivatives = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+    for i in range(len(sequence)-1):
+        if times[i+1] - times[i] == 0:
+            print "time delta is 0! skipping derivative"
+            derivative = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+        else:
+            derivative = np.array([])
+            for j in range(len(sequence[i])):
+                derivative = np.append(derivative, (sequence[i+1][j] - sequence[i][j]) / (times[i+1] - times[i]))
+        derivatives = np.vstack((derivatives, derivative))
+
+    return derivatives
+
+def get_fig_and_axes():
+    fig, axes = plt.subplots(nrows=4, ncols=6, sharex=True, sharey=True)
```

Comment:
Seems like overkill to make this a function? don't need to change though.

Reply:
Done